pgo/cmd/restart.go

```diff
@@ -0,0 +1,176 @@
+// Package cmd provides the command line functions of the crunchy CLI
+package cmd
+
+/*
+ Copyright 2020 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	msgs "github.com/crunchydata/postgres-operator/apiservermsgs"
+	"github.com/crunchydata/postgres-operator/pgo/api"
+	"github.com/crunchydata/postgres-operator/pgo/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var restartCmd = &cobra.Command{
+	Use:   "restart",
+	Short: "Restarts the PostgrSQL database within a PostgreSQL cluster",
+	Long: `Restarts one or more PostgreSQL databases within a PostgreSQL cluster.
+	
+	For example, to restart the primary and all replicas:
+	pgo restart mycluster
+
+	Or target a specific instance within the cluster:
+	pgo restart mycluster --target=mycluster-abcd
+	
+	And use the 'query' flag obtain a list of all instances within the cluster:
+	pgo restart mycluster --query`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		if OutputFormat != "" {
+			if OutputFormat != "json" {
+				fmt.Println("Error: ", "json is the only supported --output format value")
+				os.Exit(2)
+			}
+		}
+
+		if Namespace == "" {
+			Namespace = PGONamespace
+		}
+
+		if len(args) == 0 {
+			fmt.Println(`Error: You must specify the cluster to restart.`)
+		} else {
+			if Query {
+				queryRestart(args, Namespace)
+			} else if len(args) > 1 {
+				fmt.Println("Error: a single cluster must be specified when performing a restart")
+			} else if util.AskForConfirmation(NoPrompt, "") {
+				restart(args[0], Namespace)
+			} else {
+				fmt.Println("Aborting...")
+			}
```

Comment:
I think a switch statement would be nicer here.

```suggestion
			switch {
			default:
				fmt.Println("Aborting...")
			case Query:
				queryRestart(args, Namespace)
			case len(args) > 1:
				fmt.Println("Error: a single cluster must be specified when performing a restart")
			case util.AskForConfirmation(NoPrompt, ""):
				restart(args[0], Namespace)
			}
```

Reply:
Switched to a switch.